lib/mix/lib/mix/tasks/escript.ex

```diff
@@ -0,0 +1,45 @@
+defmodule Mix.Tasks.Escript do
+  use Mix.Task
+
+  @shortdoc "List installed escripts."
+
+  @moduledoc ~S"""
+  Lists all installed escripts.
+
+  Escripts are installed at `~/.mix/escripts`. Add that path to your `PATH` environment variable
+  to be able to run installed escripts from any directory.
+  """
+
+  use Bitwise
+
+  @spec run(OptionParser.argv) :: :ok
+  def run(_) do
+    escripts_path = Mix.Local.escripts_path
+
+    escripts =
+      escripts_path
+      |> list_dir()
+      |> Enum.filter(fn filename -> executable?(Path.join([escripts_path, filename])) end)
+
+    Mix.Local.Utils.print_list(escripts,
+      empty_message: "No escripts currently installed.",
+      footnote: "Escripts installed at: #{escripts_path}")
+  end
+
+  defp list_dir(path) do
+    case File.ls(path) do
+      {:ok, list} -> list
+      _ -> []
+    end
+  end
+
+  defp executable?(path) do
+    owner_exec_bit = 0o00100
+    group_exec_bit = 0o00010
+    other_exec_bit = 0o00001
+    stat = File.stat!(path)
+
+    (stat.mode &&& (owner_exec_bit ||| group_exec_bit ||| other_exec_bit)) != 0
+    and stat.type == :regular
```

Comment:
Style nitpicking: having `and` operator at the end of previous line might be better.
More context: https://github.com/elixir-lang/gettext/commit/88c6f08ab342a424f3701a88cfe1bfd2b72a8793#commitcomment-14914399.